specs/core/1_shard-data-chains.md

```diff
@@ -95,517 +94,346 @@ We define the following Python custom types for type hinting and readability:
 ```
 
-### `ShardBlockSignatures`
-
-```python
-class ShardBlockSignatures(Container):
-    attestation_signature: BLSSignature
-    proposer_signature: BLSSignature
-```
-
-### `ShardBlockCore`
+### `ShardBlockHeaderData`
 
 ```python
-class ShardBlockCore(Container):
+class ShardBlockHeaderData(Container):
     slot: ShardSlot
-    beacon_chain_root: Hash
+    beacon_block_root: Hash
     parent_root: Hash
-    data_root: Hash
     state_root: Hash
-    total_bytes: uint64
-    attester_bitfield: Bitvector[TARGET_PERSISTENT_COMMITTEE_SIZE * 2]
+    aggregation_bits: Bitvector[2 * MAX_PERIOD_COMMITTEE_SIZE]
+    block_size_sum: uint64
+    body_root: Hash
 ```
 
-### `ExtendedShardBlockCore`
+### `ShardBlockHeader`
 
 ```python
-class ExtendedShardBlockCore(Container):
-    slot: ShardSlot
-    beacon_chain_root: Hash
-    parent_root: Hash
-    data: Bytes[SHARD_BLOCK_SIZE_LIMIT - SHARD_HEADER_SIZE]
-    state_root: Hash
-    total_bytes: uint64
-    attester_bitfield: Bitvector[TARGET_PERSISTENT_COMMITTEE_SIZE * 2]
+class ShardBlockHeader(Container):
+    data: ShardBlockHeaderData
+    signatures: ShardBlockSignatures
 ```
 
 ### `ShardState`
 
 ```python
 class ShardState(Container):
-    history_accumulator: Vector[Hash, HISTORY_ACCUMULATOR_VECTOR]
-    earlier_committee_rewards: List[uint64, TARGET_PERSISTENT_COMMITTEE_SIZE]
-    later_committee_rewards: List[uint64, TARGET_PERSISTENT_COMMITTEE_SIZE]
-    earlier_committee_fees: List[Gwei, TARGET_PERSISTENT_COMMITTEE_SIZE]
-    later_committee_fees: List[Gwei, TARGET_PERSISTENT_COMMITTEE_SIZE]
-    basefee: Gwei
-    slot: ShardSlot
     shard: Shard
-    most_recent_block_core: ShardBlockCore
-    receipt_root: Hash
-    total_bytes: uint64
+    slot: ShardSlot
+    history_accumulator: Vector[Hash, HISTORY_ACCUMULATOR_VECTOR]
+    latest_block_header_data: ShardBlockHeader
+    block_size_sum: uint64
+    # Fees and rewards
+    block_size_price: Gwei
+    older_committee_fees: List[Gwei, MAX_PERIOD_COMMITTEE_SIZE]
+    newer_committee_fees: List[Gwei, MAX_PERIOD_COMMITTEE_SIZE]
+    older_committee_rewards: List[Gwei, MAX_PERIOD_COMMITTEE_SIZE]
+    newer_committee_rewards: List[Gwei, MAX_PERIOD_COMMITTEE_SIZE]
 ```
 
-### `ShardReceiptDelta`
+### `ShardCheckpoint`
 
 ```python
-class ShardReceiptDelta(Container):
-    index: ValidatorIndex
-    reward_coefficient: uint64
-    block_fee: Gwei
+class ShardCheckpoint(Container):
+    slot: ShardSlot
+    parent_root: Hash
 ```
 
 ## Helper functions
 
-### `compute_slot_of_shard_slot`
-
-```python
-def compute_slot_of_shard_slot(slot: ShardSlot) -> Epoch:
-    return Epoch(slot // SHARD_SLOTS_PER_BEACON_SLOT)
-```
+### Misc
 
-### `compute_epoch_of_shard_slot`
+#### `compute_epoch_of_shard_slot`
 
 ```python
 def compute_epoch_of_shard_slot(slot: ShardSlot) -> Epoch:
-    return Epoch(slot // SHARD_SLOTS_PER_BEACON_SLOT // SLOTS_PER_EPOCH)
+    return compute_epoch_of_slot(slot // SHARD_SLOTS_PER_EPOCH)
```

Comment:
```suggestion
    return Epoch(slot // SHARD_SLOTS_PER_EPOCH)
```

Reply:
`compute_epoch_of_slot` is correct here. Might need a cast of `slot // SHARD_SLOTS_PER_EPOCH` to `Slot` before being passed into `compute_epoch_of_slot`

Reply:
Ahh I see it now, tricky...

Reply:
That sounds like it's dividing by epoch length squared....

Maybe `return compute_epoch_of_slot(slot // SHARD_SLOTS_PER_BEACON_SLOT)` ?

Reply:
I think @terencechain is right. Fixed it to `return Epoch(slot // SHARD_SLOTS_PER_EPOCH)`.